nemo/collections/asr/parts/jasper.py

```diff
@@ -71,7 +71,7 @@ def compute_new_kernel_size(kernel_size, kernel_width):
 def get_same_padding(kernel_size, stride, dilation) -> int:
     if stride > 1 and dilation > 1:
         raise ValueError("Only stride OR dilation may be greater than 1")
-    return (dilation * (kernel_size - 1)) // 2
+    return (kernel_size + (kernel_size - 1) * (dilation - 1) - 1) // 2
```

Comment:
This formula is the exact same formula as the old one. Just expand:
`k + (k - 1) * (d - 1) - 1 = k + kd - k - d  + 1 - 1 = kd - d = d * (k - 1)`

You can also verify numerically:
```
def p1(k, d):
  return (d * (k - 1))//2

def p2(k ,d):
   return (k + (k - 1) * (d - 1) - 1) // 2


for k in range(1, 1001):
   for d in range(1, 1000):
     if p1(k, d) != p2(k, d):
       print(k, d)
```

I would suggest keeping the old formula and just adding tests in this PR.

Reply:
@nithinraok The new formula here and the one before are identical. The expanded formula is far harder to interpret and extend to the asymmetric case as well. Revert and just add the tests. As a sanity check, also add the above code snippet as a test.

Reply:
Sorry was basing my PR on my local changes from rc1 branch which has a previous formula of `(d*k)//2 -1` didn't notice it was fixed in main. changed the formula to its shorter form of `d*(k-1)` funny that I missed it. Will update with tests only